TrackerApp/Trackers/Categories-MVVM/View/CategoriesViewController.swift

```diff
@@ -75,36 +76,37 @@ final class CategoriesViewController: UIViewController {
         categoriesTableView.isHidden = viewModel.categories.isEmpty
     }
 
-    func initialize(_ viewModel: CategoriesViewModel) {
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        guard let viewModel = viewModel, !viewModel.categories.isEmpty else { return }
+        categoriesTableView.scrollToRow(at: viewModel.selectedCategoryIndexPath, at: .bottom, animated: true)
+    }
+
+    init(viewModel: CategoriesViewModelProtocol) {
+        super.init(nibName: nil, bundle: nil)
         self.viewModel = viewModel
-        bind()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
     }
 
     private func bind() {
         guard let viewModel = viewModel else { return }
         viewModel.categoriesObservable.bind { [weak self] _ in
+            self?.categoriesTableView.isHidden = viewModel.categories.isEmpty
             self?.categoriesTableView.reloadData()
+            self?.dismissWithCallback()
         }
     }
 
     @objc private func addCategoryButtonAction() {
+        newCategoryViewController = NewCategoryViewController()
+        guard let newCategoryViewController = newCategoryViewController else { return }
```

Comment:
можно же проще:

```suggestion
        let newCategoryViewController = NewCategoryViewController()
```

Это все еще относится к моему комментарию про сохранение ссылок)